src/v/cloud_topics/level_zero/stm/ctp_stm_state.h

```diff
@@ -134,6 +134,10 @@ class ctp_stm_state
     /// This is used to lookup the epoch that was last reconciled for
     /// L0 GC.
     std::optional<model::offset> _last_reconciled_log_offset;
+
+    // The last observed epoch to be applied to the state machine. This value is
+    // used to check for violations of monotonicity in epoch order.
+    cluster_epoch _last_seen_epoch{};
```

Comment:
I think that `_max_applied_epoch` could be used instead.

Reply:
> I think that _max_applied_epoch could be used instead.

i tried, but it wasn't obvious. especially since there was some other place where it was updated.